pathauto.install

```diff
@@ -51,3 +52,137 @@ function pathauto_update_8001() {
+  }
+
+  $messages = array();
+  $entity_manager = \Drupal::service('entity.manager');
+  $entity_type_manager = \Drupal::service('entity_type.manager');
+  $language_manager = \Drupal::service('language_manager');
+  $entity_info = $entity_manager->getDefinitions();
+
+  // 1. Load all patterns.
+  $config = \Drupal::service('config.factory')->getEditable('pathauto.pattern');
+  $patterns = $config->get('patterns');
+
+  // 2. Create a configuration entity per pattern.
+  foreach ($patterns as $entity_type => $entity_patterns) {
+    if (!array_key_exists($entity_type, $entity_info)) {
+      // We found an unknown entity type. Report it.
+      $messages[] = t('Entity of type @type was not processed. It defines the following patterns: @patterns', array(
+        '@type' => $entity_type,
+        '@patterns' => print_r($entity_patterns, TRUE),
+      ));
+      continue;
+    }
+    $entity_label = $entity_info[$entity_type]->getLabel();
+
+    if (isset($entity_patterns['default'])) {
+      // This is a pattern for an entity type, such as "node".
+      $entity = entity_create('pathauto_pattern', [
+        'id' => $entity_type,
+        'label' => $entity_label,
+        'type' => 'canonical_entities:' . $entity_type,
+        'pattern' => $entity_patterns['default'],
+        'weight' => 0,
+      ]);
+      $entity->save();
+    }
+
+    // Loop over bundles and create patterns if they have a value.
+    // Bundle keys may have a language suffix for language-dependant patterns.
+    if (isset($entity_patterns['bundles'])) {
+      $bundle_info = $entity_manager->getBundleInfo($entity_type);
+      foreach ($entity_patterns['bundles'] as $bundle => $bundle_patterns) {
+        if (empty($bundle_patterns['default'])) {
+          // This bundle does not define a pattern. Move on to the next one.
+          continue;
+        }
+
+        if (isset($bundle_info[$bundle])) {
+          // This is a pattern for a bundle, such as "node_article".
+          $entity = entity_create('pathauto_pattern', [
+            'id' => $entity_type . '_' . $bundle,
+            'label' => $entity_label . ' ' . $bundle_info[$bundle]['label'],
+            'type' => 'canonical_entities:' . $entity_type,
+            'pattern' => $bundle_patterns['default'],
+            'weight' => -5,
+          ]);
+
+          // Add the bundle condition.
+          $entity->addSelectionCondition([
+            'id' => 'entity_bundle:' . $entity_type,
+            'bundles' => array($bundle),
+            'negate' => FALSE,
+            'context_mapping' => [ $entity_type => $entity_type ],
+          ]);
+
+          $entity->save();
+        }
+        else {
+          // This is either a language dependent pattern such as "article_es" or
+          // an unknown bundle or langcode. Let's figure it out.
+          $matches = NULL;
+          $langcode = NULL;
+          preg_match('/^(.*)_([a-z-]*)$/', $bundle, $matches);
```

Comment:
Does this work with a bundle like some_thing? I guess so.. Instead of *, maybe use + so that you require at least one character.